src/core_init_atmosphere/mpas_init_atm_gwd.F

```diff
@@ -305,48 +334,231 @@ function compute_gwd_fields(domain) result(iErr)
          end if
       end do
 
-
-      deallocate(topo)
-      deallocate(landuse)
       deallocate(hlanduse)
 
+      iErr = remove_tiles(tilesHead)
+
```

Comment:
Note that the line immediately following sets `iErr = 0`, masking any error returned by `remove_tiles`.

Reply:
removing the `iErr = 0` below this